nav-app/src/app/data.service.ts

```diff
@@ -327,7 +327,10 @@ export abstract class BaseStix {
         this.id = stixSDO.id;
         this.name = stixSDO.name;
         this.description = stixSDO.description;
-        this.attackID = stixSDO.external_references[0].external_id;
+        if (stixSDO.external_references && stixSDO.external_references[0]) this.attackID = stixSDO.external_references[0].external_id; else {
+          alert('Error: external_id in external_references not found in imported BaseStix object id (' + stixSDO.id + ')');
+          throw console.error('Error: external_id in external_references not found in imported BaseStix object', stixSDO, ' Read more here: https://docs.oasis-open.org/cti/stix/v2.1/os/stix-v2.1-os.html#_72bcfr3t79jx');
```

Comment:
Is `console.error()` a valid thing to throw? I would think you'd throw a string or `new Error('desc')` here.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw#description

Reply:
I believe it would in this case throw the value returned by calling console.error, which is presumably null/undefined.